bar_app/serializers.py

```diff
@@ -0,0 +1,169 @@
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+from django.contrib.auth.models import User
+from .models import *
+
+class UserSerializer(serializers.ModelSerializer):
+    """
+    Serializer for User model.
+    """
+    class Meta:
+        model = User
+        fields = ["username", "password"]
+        extra_kwargs = {"password": {"write_only": True}}
+
+class ReferenceSerializer(serializers.ModelSerializer):
+    """
+    Serializer for Reference model.
+    """
+    availability = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Reference
+        fields = "__all__"
+        read_only_fields = ["id"]
+
+    def validate_ref(self, value):
+        """
+        Check if the ref exists in the database.
+        """
+        if Reference.objects.filter(ref=value).exists():
+            raise serializers.ValidationError("This reference already exists.")
+        return value
+
+    def validate_name(self, value):
```

Comment:
this is useless and already done by DRF. can you explain me why/how ?

Reply:
Same for the previous comment